assets/blocks/core-image/index.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * External dependencies
+ */
+import { addFilter } from '@wordpress/hooks';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { Spinner } from '@wordpress/components';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import './style.scss';
+
+/**
+ * Add image credit meta to core/image block attributes
+ */
+addFilter(
+	'blocks.registerBlockType',
+	'newspack-plugin/register-hook/core-image',
+	( settings, name ) => {
+		if ( name !== 'core/image' ) {
+			return settings;
+		}
+		return {
+			...settings,
+			attributes: {
+				...settings.attributes,
+				meta: {
+					type: 'object',
+					default: {},
+				},
+			},
+		};
+	}
+);
+
+/**
+ * Display spinner and lock post saving until meta attributes are added to block
+ */
+const AttributesLoader = props => {
```

Comment:
With `woocomerce` plugin active, this component causes an error to be thrown:

<img width="775" alt="image" src="https://github.com/Automattic/newspack-plugin/assets/7383192/12cd0cf7-9ded-492d-a9cb-9213502bb376">

Reply:
This was automatically fixed with https://github.com/Automattic/newspack-plugin/pull/2936/commits/3142c27b72185fa0e54677b27abba52bd86791c2